### DISCUSSION TITLE
Why do we need to wrap the Calendar with Popover and Dialog components?

### DISCUSSION THREAD

**Discussion by ArtemFedorchuk** [adobe/react-spectrum#4692] [General]
Is somebody has an idea why do we need to wrap the Calendar with Popover and Dialog components?
What is the idea and what does this solve?

<img width="724" alt="Screenshot 2023-06-22 at 18 19 34" src="https://github.com/adobe/react-spectrum/assets/43110065/fd5efca7-17c4-405a-be7d-bbc896d012e2">

**Comment by LFDanLu** [comment]
The Popover makes it so the contents are rendered in a overlay above other components in your app. It comes with focus restoration to the trigger when the Popover is closed, various accessibility benefits like hiding elements outside the Popover from screen readers, and handles placement of the contents relative to the trigger you specify. The Dialog then is the container around the Calendar and enables focus containment within the Popover.

**Reply by ArtemFedorchuk** [reply]
Thanks
